webapp/src/ts/components/filters/analytics-filter/analytics-filter.component.ts

```diff
@@ -30,9 +65,67 @@ export class AnalyticsFilterComponent implements AfterContentInit, AfterContentC
     this.subscriptions.unsubscribe();
   }
 
+  private subscribeToStore() {
+    const subscription = this.store.select(Selectors.getSidebarFilter).subscribe(({ isOpen }) => {
+      this.isOpen = isOpen;
+    });
+    this.subscriptions.add(subscription);
+  }
+
   private setActiveModule() {
     this.activeModule = this.analyticsModules?.find(module => {
       return module.id === this.route.snapshot?.firstChild?.data?.moduleId;
     });
   }
+
+  private subscribeToRouteChanges() {
+    const routeSubscription = this.router.events.pipe(
+      filter(event => event instanceof NavigationEnd)
+    ).subscribe(() => {
+      this.canDisplayFilterButton();
+    });
+    this.subscriptions.add(routeSubscription);
+  }
+
+  private hasMultipleFacilities() {
+    return this.userSettingsService
+      .get()
+      .then((userSettings: any) => {
+        const facility = userSettings.facility_id;
+        return Array.isArray(facility) && facility.length > 1;
+      });
+  }
+
+  private checkPermissions() {
+    return Promise
+      .all([
+        this.authService.has('!can_view_old_filter_and_search'),
+        this.authService.has('!can_view_old_action_bar')
+      ])
+      .then(([cannotViewOldFilter, cannotViewOldActionBar]) => {
+        return cannotViewOldFilter && cannotViewOldActionBar;
+      })
+      .catch(error => {
+        console.error('Error checking permissions:', error);
+        return false;
+      });
+  }
+
+  private isTargetAggregates() {
+    return this.router.url.startsWith(AGGREGATE_TARGET_URL);
```

Comment:
1. You can keep the constant at this level instead of outside the class, because it's only used here.  
2. and check by the `moduleId` that is part of the router's data object. 

<img width="508" alt="Screenshot 2024-07-09 at 11 26 01 AM" src="https://github.com/medic/cht-core/assets/66472237/168091bc-07c1-47b5-b3ae-073d4b759096">

3. That is already available in this component. See `setActiveModule()` in L77, so you can reuse that check.